component.yaml

```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: v1
+kind: Component
+spec:
+  type: scheduler
```

Comment:
Re. name generation: It does not appear necessary to introduce a new name generation mechanism to the API. It looks like generateName will work.